src/bridge/cockpitpeer.c

```diff
@@ -96,15 +96,23 @@ G_DEFINE_TYPE (CockpitPeer, cockpit_peer, G_TYPE_OBJECT);
 
 static void
 startup_done (CockpitPeer *self,
-              const gchar *problem)
+              const gchar *problem,
+              const gchar *stderr)
```

Comment:
I feel like we could increase what this function is doing:

1) drop the `stderr` parameter again
2) if `problem != NULL` then call `take_stderr()` ourselves
3) always stop stderr here

That could probably remove some of the extra handling code that is happening at various call sites below.

Reply:
There is one case where we destroy `self->other` before calling `startup_done` (and we need to get stderr before the destruction). I see if that can be made to work anyway.

Reply:
Seems to work, thanks for the suggestion!